Extension/src/Utility/Eventing/dispatcher.ts

```diff
@@ -0,0 +1,502 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/unified-signatures */
+
+import { readFile } from 'node:fs/promises';
+import { ManualPromise } from '../Async/manualPromise';
+import { ManualSignal } from '../Async/manualSignal';
+import { returns } from '../Async/returns';
+import { filepath } from '../Filesystem/filepath';
+import { hasErrors } from '../Sandbox/interfaces';
+import { Sandbox } from '../Sandbox/sandbox';
+import { collectGarbage } from '../System/garbageCollector';
+import { is } from '../System/guards';
+import { isAnonymousObject, members, typeOf } from '../System/info';
+import { getOrAdd } from '../System/map';
+import { verbose } from '../Text/streams';
+import { Descriptors } from './descriptor';
+import { parse } from './eventParser';
+import { ArbitraryObject, Callback, Continue, EventData, EventStatus, Subscriber, Subscription, Unsubscribe } from './interfaces';
+
+interface Event<TInput = any, TResult = void> {
+    readonly name: string;
+    readonly descriptors: Descriptors;
+
+    completed?: ManualPromise<EventStatus | TResult>;
+    readonly variableArgs: any[];
+    source?: ArbitraryObject | undefined;

```

Comment:
Can you remove the `| undefined` from the 3 members? Our existing code doesn't use that with `?:`.